### PR TITLE
concord-server: store process dependencies in DB

### PR DESCRIPTION
**Pull request by brig**

### CODE REVIEW

File: runtime/loader/src/main/java/com/walmartlabs/concord/process/loader/v2/ConfigurationV2.java [hunk 1]
```diff
@@ -27,22 +27,31 @@
...
+
         ObjectMapper om = new ObjectMapper();
         om.registerModule(new Jdk8Module());
         om.registerModule(new JavaTimeModule());
         this.values = om.convertValue(cfg, Map.class);
     }
 
+    @Override
+    public List<String> dependencies() {
+        return cfg.dependencies();
```
Comment (ibodrov) [inline runtime/loader/src/main/java/com/walmartlabs/concord/process/loader/v2/ConfigurationV2.java:52]: Why not to use the `values` object like in `ConfigurationV1`?
This way we can avoid saving a reference to `ProcessConfiguration` and duplicated data if someone serializes a `ConfigurationV2` instance.

File: server/db/src/main/resources/com/walmartlabs/concord/server/db/v1.60.0.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+
+    <changeSet id="1600000" author="ybrigo@gmail.com">
+        <addColumn tableName="PROCESS_QUEUE">
+            <column name="DEPENDENCIES" type="varchar[]">
```
Comment (ibodrov) [inline server/db/src/main/resources/com/walmartlabs/concord/server/db/v1.60.0.xml:9]: `text[]` maybe? Like in the previous change sets?

**Comment by ibodrov** [comment]
@brig Do we need a hard limit on the number of dependencies? To prevent someone from putting a million lines into `dependencies` and, subsequently, into the DB.

File: runtime/loader/src/main/java/com/walmartlabs/concord/process/loader/v2/ConfigurationV2.java (hunk 1 above)
Comment (brig) [reply runtime/loader/src/main/java/com/walmartlabs/concord/process/loader/v2/ConfigurationV2.java:52]: fixed

File: server/db/src/main/resources/com/walmartlabs/concord/server/db/v1.60.0.xml (hunk 2 above)
Comment (brig) [reply server/db/src/main/resources/com/walmartlabs/concord/server/db/v1.60.0.xml]: fixed

**Pull request merged**